### PR TITLE
am pus tema selenium

### PR DESCRIPTION
**Pull request by Cosmin0589**

### CODE REVIEW

**Review by 95teo — CHANGES_REQUESTED** [review]
Let's try to make a call these days. It looks like you've created an entire new maven project and it's not necessary. We only need the java classes. Thanks!:D

File: src/test/java/cosminsechel/tests/seleniumpart1/TablePageTests.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        List<Integer> age = new ArrayList<>();
+        age.add(1);
+        age.add(2);
+        age.add(3);
+
+        String jsonFormated = tablePage.createJson(name,age);
+        System.out.println(jsonFormated);
+        tablePage.changeTableData(jsonFormated);
+
+
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/TablePageTests.java:37]: Leave only one blank line here. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]
You are missing the other 3 pages from https://testpages.herokuapp.com/styled/index.html and also you have nothing for https://demoqa.com/elements. Please read again the homework requirements. Thanks!

File: src/main/java/cosminsechel/driver/BrowserManager.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+public class BrowserManager {
+
+    public static ChromeDriver createDriver() {
+        WebDriverManager.chromedriver().setup();
+        return new ChromeDriver();
+    }
+
+    public static void closeChromeDriver(ChromeDriver driver) {
+    }
```
Comment (95teo) [inline src/main/java/cosminsechel/driver/BrowserManager.java:14]: You forgot to add driver.quit() inside this method. Don't leave the method body empty. Thanks!

File: src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import cosminsechel.driver.BrowserManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class Demoqatema {
+
+    public static void main(String[] args) throws InterruptedException {
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java:11]: You are opening multiple browser instances. We only need one.

File: src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        BrowserManager.createDriver();
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java:12]: Remove this line. Thanks!

File: src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        WebDriver driver = new ChromeDriver();
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java:13]: This line should be using the BrowserManager.createDriver() method -> update existing line to ->
WebDriver driver = BrowserManager.createDriver();

File: src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        driver.findElement(By.id("item-1")).click();
+        WebElement checkbox = driver.findElement(By.cssSelector("label[for='tree-node-home']"));
+        checkbox.click();
+        if (checkbox.isSelected()) {
+            System.out.println("checkbox selected");
+
+        }else{
+            System.out.println("checkbox not selected ");
+        }
+
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java:27]: Please close the browser at the end. You can use the method defined in BrowserManager.

File: src/test/java/cosminsechel/tests/seleniumpart1/Test.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+import cosminsechel.driver.BrowserManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class Test {
+    public static void main(String[] args ) throws InterruptedException {
+        BrowserManager.createDriver();
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Test.java:11]: Same comments as in the previous class. Thanks!

File: src/test/java/cosminsechel/tests/seleniumpart1/Test.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        WebElement p1 = driver.findElement(By.id("para1"));
+        System.out.println(p1.getText());
+        WebElement p2 = driver.findElement(By.id("para2"));
+        System.out.println(p2.getText());
+        driver.navigate().to("https://testpages.herokuapp.com/styled/attributes-test.html");
+        WebElement p2page = driver.findElement(By.xpath("/html/body/div/div[2]/child::*"));
+        System.out.println(p2page.getText());
+        WebElement button = driver.findElement(By.className("styled-click-button"));
+        button.click();
+
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Test.java:42]: You can remove these two blank lines from here as they are not needed. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by sorinfoca — APPROVED** [review]

File: src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        driver.findElement(By.id("userName")).sendKeys("Cosmin");
+        driver.findElement(By.id("item-1")).click();
+        WebElement checkbox = driver.findElement(By.cssSelector("label[for='tree-node-home']"));
+        checkbox.click();
+        if (checkbox.isSelected()) {
+            System.out.println("checkbox selected");
+
+        }else{
+            System.out.println("checkbox not selected ");
+            closeBrowser();
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java:24]: Move the closeBrowser(); call after if - else block. We want to close always the browser, not only if we're not selecting the checkbox. Thanks!

File: src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+
+    }
+
+    private static void closeBrowser() {
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Demoqatema.java:30]: You can remove this method from here as you defined the closeBrowser in BrowserManager class. That's the method you want to use to close the browser.

File: src/test/java/cosminsechel/tests/seleniumpart1/Test.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+import cosminsechel.driver.BrowserManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class Test {
+    public static void main(String[] args ) throws InterruptedException {
+        BrowserManager.createDriver();
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Test.java:11]: You forgot to delete this line so please remove it. Thanks!

File: src/test/java/cosminsechel/tests/seleniumpart1/Test.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+        WebElement p1 = driver.findElement(By.id("para1"));
+        System.out.println(p1.getText());
+        WebElement p2 = driver.findElement(By.id("para2"));
+        System.out.println(p2.getText());
+        driver.navigate().to("https://testpages.herokuapp.com/styled/attributes-test.html");
+        WebElement p2page = driver.findElement(By.xpath("/html/body/div/div[2]/child::*"));
+        System.out.println(p2page.getText());
+        WebElement button = driver.findElement(By.className("styled-click-button"));
+        button.click();
+    }
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/seleniumpart1/Test.java:42]: Add BrowserManager.closeChromeDriver(driver) at the end to close the browser. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by 95teo — APPROVED** [review]

**Pull request merged**